commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -369,6 +369,28 @@ def get_payment_by_transaction_interaction_id(self, interaction_id: str) -> CTPa
         logger.info(f"[CommercetoolsAPIClient] - Attempting to find payment with interaction ID {interaction_id}")
         return self.base_client.payments.query(where=f'transactions(interactionId="{interaction_id}")').results[0]
 
+    def get_product_variant_by_program(self, product_id: str) -> Optional[CTProductVariant]:
+        """
+        Fetches a program from Commercetools.
+        Args:
+            product_id: The ID of the program (bundle) to fetch.
+        Returns:
+            CTProductVariant if found, None otherwise.
+        """
+        start_time = datetime.datetime.now()
+        try:
+            product_projection = self.base_client.product_projections.get_by_key(product_id)
+        except CommercetoolsError as exc:
+            logger.exception(
+                f'[get_product_variant_by_program] Failed to get CT program for product_id: {product_id} '
+                f'with exception: {exc}'
+            )
+            return None
```

Comment:
Please remove this try/except here (see examples of the other client method `get_product_variant_by_course_run `) and encapsulate this method's call in a try/except block in commerce_coordinator/apps/rollout/pipeline.py file because we also want to log and redirect users to the legacy is no product is found, or API call raises an exception

See example: https://github.com/edx/commerce-coordinator/blob/2670c6303f319c79725747b8e08c7baccb70a206/commerce_coordinator/apps/rollout/pipeline.py#L46-L54